src/mainviews/list-jobs.tsx

```diff
@@ -148,38 +191,77 @@ export function NotebookJobsListBody(
     }
   ];
 
+  const rows: JSX.Element[] = notebookJobs.jobs
+    .filter(job => !deletedRows.has(job.job_id))
+    .map(job =>
+      buildTableRow(
+        job,
+        props.app,
+        props.showCreateJob,
+        deleteRow,
+        translateStatus
+      )
+    );
```

Comment:
Building all JSX elements here, but only using a slice of data at the bottom. Would there be a way to do the slicing of data first and only use JSX rows for the sliced data?

Reply:
Again, great comment. Fixed.